### ISSUE TITLE
Use CI to generate zip list

### ISSUE THREAD

**Issue by defisym** [NaitorStudios/FusionShaderRepo#1]
Currently all files are bundled in one zip file, it's not that easy to view sources and contribute. 
Personally I suggest to use github action for continuous integration, for example, by using the following steps: 

- create a new github.io repo for release
- Upload compile tool to this repo, for later use
- set an on push action, uses windows
- clone repo
- compile shaders, if no error, continue (can be removed together with step 2 if all contributors provides compiled files)
- compress all shader files separately to a temp folder
- generate file list page then push to the release folder

Reference:

https://help.github.com/en/articles/configuring-a-workflow

https://github.com/marketplace/actions/action-zip

https://github.com/JamesIves/github-pages-deploy-action

Thanks in advance.

**Comment by NaitorStudios** [comment]
I'm not sure how to use any of these, I barely know how to use Github, just picked it to get direct download links.
I would originally use Google Drive. So I think you misunderstood the purpose of this...
This was commissioned to be used on Darkwire's new shader list, I've made the list with all shader's information.
These direct links will be provided there, I don't know any other way to not have it zipped and still get a zip direct download link...

In resume, I need direct individual ZIP downloads for Darkwire.
It would be nice to have the source code nicely shown and people be able to contribute but I'm not sure how to do it.
If you know a way, please let me know.

**Comment by defisym** [comment]
here is a solution:

1. generate a token
![image](https://user-images.githubusercontent.com/55148235/217579892-85116b32-8124-4d2d-81a6-9a65ef359fb6.png)
2. add a secret in repo
![image](https://user-images.githubusercontent.com/55148235/217580069-a61ae011-8aaa-4153-857d-8730687e1f8a.png)
3. upload tools, e.g. 7zip & bat, will use it later
![image](https://user-images.githubusercontent.com/55148235/217580272-a5e8dba3-fe95-4ddf-823a-4b6db49fe94c.png)
3.1. a ref of bat which zips all files then put into release folder
you can also use [compile tools](https://github.com/defisym/OpenFusionExamples/tree/master/Tools/DX11%20Shader%20Compile%20Tool) to auto compile it, just in the `JobPath` loop
```bat
@REM @echo off

Set CompressPath=%~dp07z.exe

Set JobPath=%~dp0..\Files
Set ReleasePath=%~dp0..\Release

for /r "%ReleasePath%" %%F in (*.*) do (
	del "%%F" /f /s /q	
)

CD /d %JobPath%

for /f %%i in ('dir /b /ad "%JobPath%"') do (
	echo %%i
    %CompressPath% a -tzip %ReleasePath%\%%i.zip %JobPath%\%%i\*
)

```
4. create a workflow, this will add new files then commit and push
```yml
# This is a basic workflow to help you get started with Actions

name: CI

# Controls when the workflow will run
on:
  # Triggers the workflow on push or pull request events but only for the "main" branch
  push:
    branches: [ "main" ]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: windows-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v3    
        with:
          token: ${{ secrets.CI_TOKEN }}    

      # Runs a single command using the runners shell
      - name: Run a one-line script
        working-directory: ${{env.GITHUB_WORKSPACE}}
        run: ./tools/zip.bat

      # Update references
      - name: Commit update
        working-directory: ${{env.GITHUB_WORKSPACE}}
        run: |          
          git config --global user.name 'Git bot'
          git config --global user.email 'bot@noreply.github.com'
          git add .
          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
          git commit -am "Auto updated" && git push || echo "No changes to commit"
```

if you want to put zips in another repo, then you can follow the steps of https://github.com/JamesIves/github-pages-deploy-action and modify the conterpart of the yml

5. done, github will auto pack all your shaders to zip once you pushed.
![image](https://user-images.githubusercontent.com/55148235/217582783-cc344c1c-de39-4175-9364-bb3d84d431c6.png)

**Comment by NaitorStudios** [comment]
Hm I'll try, but for now I've added all effects unpacked as well, in case people just want to check the code or contribute with updates.
